File: app/src/main/java/com/example/daypilot_test_desing/core/data/repository/ReminderRepository.kt
package com.example.daypilot_test_desing.core.data.repository

import com.example.daypilot_test_desing.core.data.model.ReminderData
import com.example.daypilot_test_desing.core.data.model.ReminderFormDataInfo

interface ReminderRepository {
    fun getReminders(): List<ReminderData>
    fun addReminder(form: ReminderFormDataInfo): ReminderData
    fun deleteReminder(id: String)
    fun toggleReminder(id: String, enabled: Boolean)
}

File: lib/core/data/repositories/reminder_repository.dart
import '../models/app_reminder.dart';

abstract class ReminderRepository {
  Future<List<AppReminder>> getReminders();
  Future<AppReminder> addReminder(AppReminder reminder);
  Future<void> deleteReminder(String id);
  Future<void> toggleReminder(String id, bool enabled);
  Future<void> updateDateTime(String id, DateTime dateTime);
}
